firefox-ios/Client/Frontend/Onboarding/Models/TermsOfServiceTelemetry.swift

```diff
@@ -6,45 +6,52 @@ import Foundation
 import Glean
 
 struct TermsOfServiceTelemetry {
+    private let gleanWrapper: GleanWrapper
+
+    init(gleanWrapper: GleanWrapper = DefaultGleanWrapper()) {
+        self.gleanWrapper = gleanWrapper
+    }
+
     func termsOfServiceScreenDisplayed() {
-        GleanMetrics.Onboarding.termsOfServiceCard.record()
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.termsOfServiceCard)
     }
 
     func technicalInteractionDataSwitched(to value: Bool) {
         let extra = GleanMetrics.Onboarding.ToggleTechnicalInteractionDataExtra(changedTo: value)
-        GleanMetrics.Onboarding.toggleTechnicalInteractionData.record(extra)
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.toggleTechnicalInteractionData, extras: extra)
     }
 
     func automaticCrashReportsSwitched(to value: Bool) {
         let extra = GleanMetrics.Onboarding.ToggleAutomaticCrashReportsExtra(changedTo: value)
-        GleanMetrics.Onboarding.toggleAutomaticCrashReports.record(extra)
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.toggleAutomaticCrashReports, extras: extra)
     }
 
     func termsOfServiceLinkTapped() {
-        GleanMetrics.Onboarding.termsOfServiceLinkClicked.record()
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.termsOfServiceLinkClicked)
     }
 
     func termsOfServicePrivacyNoticeLinkTapped() {
-        GleanMetrics.Onboarding.termsOfServicePrivacyNoticeLinkClicked.record()
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.termsOfServicePrivacyNoticeLinkClicked)
     }
 
     func termsOfServiceManageLinkTapped() {
-        GleanMetrics.Onboarding.termsOfServiceManageLinkClicked.record()
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.termsOfServiceManageLinkClicked)
     }
 
     func termsOfServiceAcceptButtonTapped(acceptedDate: Date) {
-        GleanMetrics.Onboarding.termsOfServiceAccepted.record()
+        gleanWrapper.recordEvent(for: GleanMetrics.Onboarding.termsOfServiceAccepted)
         recordDateAndVersion(acceptedDate: acceptedDate)
     }
 
     func recordDateAndVersion(acceptedDate: Date) {
-        // Record the ToU version and date metrics with onboarding surface
+        let termsOfUseVersion: Int64 = 4
         let acceptedExtra = GleanMetrics.TermsOfUse.AcceptedExtra(
-            surface: TermsOfUseTelemetry.Surface.onboarding.rawValue,
-            touVersion: String(TermsOfUseTelemetry().termsOfUseVersion)
+            surface: "onboarding",
+            touVersion: String(termsOfUseVersion)
```

Comment:
Let's keep these using the original constants ( `ermsOfUseTelemetry.Surface.onboarding.rawValue` and `TermsOfUseTelemetry().termsOfUseVersion` ) so we only have one source of truth. If they ever change in that file we risk having the telemetry version for instance different from the other version.